mod-source-record-manager-server/src/main/java/org/folio/services/journal/InvoiceUtil.java

```diff
@@ -0,0 +1,143 @@
+import org.folio.rest.jaxrs.model.JournalRecord;
+import org.folio.rest.jaxrs.model.Record;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+
+import static org.apache.commons.lang3.StringUtils.isAnyEmpty;
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+import static org.folio.dao.JobExecutionSourceChunkDaoImpl.LOGGER;
+import static org.folio.rest.jaxrs.model.DataImportEventTypes.DI_ERROR;
+import static org.folio.rest.jaxrs.model.EntityType.EDIFACT_INVOICE;
+import static org.folio.rest.jaxrs.model.JournalRecord.*;
+import static org.folio.rest.jaxrs.model.JournalRecord.EntityType.INVOICE;
+import static org.folio.services.journal.JournalUtil.ERROR_KEY;
+
+public class InvoiceUtil {
+
+  public static final String INVOICE_LINES_KEY = "INVOICE_LINES";
+  public static final String INVOICE_LINES_ERRORS_KEY = "INVOICE_LINES_ERRORS";
+
+  public static final String FIELD_DESCRIPTION = "description";
+  public static final String FIELD_FOLIO_INVOICE_NO = "folioInvoiceNo";
+  public static final String FIELD_ID = "id";
+  public static final String FIELD_INVOICE_LINE_NUMBER = "invoiceLineNumber";
+  public static final String FIELD_INVOICE_LINES = "invoiceLines";
+  public static final String FIELD_INVOICE_NO = "invoiceNo";
+  public static final String FIELD_SOURCE_ID = "sourceId";
+  public static final String FIELD_VENDOR_INVOICE_NO = "vendorInvoiceNo";
+
+  public static final String JOURNAL_RECORD = "journalRecord";
+  private static final String EVENT_HAS_NO_DATA_MSG = "Failed to handle %s event, because event payload context does not contain %s and/or %s and/or %s data";
+
+  public static LinkedList<JournalRecord> buildJournalRecordByEvent(DataImportEventPayload event)
+    throws JournalRecordMapperException, JsonProcessingException {
+    String edifactAsString = event.getContext().get(EDIFACT_INVOICE.value());
+    String invoiceAsString = event.getContext().get(INVOICE.value());
+    String invoiceLinesAsString = event.getContext().get(INVOICE_LINES_KEY);
+
+    if (isAnyEmpty(edifactAsString, invoiceAsString, invoiceLinesAsString)) {
+      event.getContext().keySet().forEach(key -> {
+        if (isNotEmpty(event.getContext().get(key))) {
+          LOGGER.error(key + ": " + event.getContext().get(key));
+        }
+      });
+      throw new JournalRecordMapperException(String.format(EVENT_HAS_NO_DATA_MSG, event.getEventType(),
+        EDIFACT_INVOICE.value(), INVOICE.value(), INVOICE_LINES_KEY));
+    }
+
+    LinkedList<JournalRecord> journalRecords = new LinkedList<>();
+    Map<String, Object> journalInvoiceRecord = buildInvoiceRecord(event);
+    journalRecords.add((JournalRecord) journalInvoiceRecord.get(JOURNAL_RECORD));
+    journalRecords.addAll(buildInvoiceLineRecords(event, (String) journalInvoiceRecord.get(FIELD_INVOICE_NO),
+      (String) journalInvoiceRecord.get(FIELD_SOURCE_ID), event.getContext().containsKey(ERROR_KEY)));
```

Comment:
we always have an error message at the payload context when `eventType == DI_ERROR` because it is set by EventManager in di-core library

we can check for `event.getContext().containsKey(INVOICE_LINES_ERRORS_KEY)` - isInvoiceLinesLevelError
`eventType == DI_ERROR` and `event.getContext().containsKey(INVOICE_LINES_ERRORS_KEY) == false` - an error at the invoice level;
`eventType == DI_ERROR` and `event.getContext().containsKey(INVOICE_LINES_ERRORS_KEY) == true` - an error at the invoice lines level

Reply:
Fixed.